catalog/harvest/catalog-harvest-persistentlistener/src/main/java/org/codice/ddf/catalog/harvest/listeners/PersistentListener.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.harvest.listeners;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.catalog.harvest.HarvestException;
+import org.codice.ddf.catalog.harvest.HarvestedResource;
+import org.codice.ddf.catalog.harvest.Listener;
+import org.codice.ddf.catalog.harvest.StorageAdapter;
+import org.codice.ddf.catalog.harvest.common.FileSystemPersistenceProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PersistentListener implements Listener {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PersistentListener.class);
+
+  private final StorageAdapter adapter;
+
+  private final FileSystemPersistenceProvider persistenceProvider;
+
+  public PersistentListener(StorageAdapter adapter, String pid) {
+    this.adapter = adapter;
+    persistenceProvider =
+        new FileSystemPersistenceProvider("harvest/persistent/" + DigestUtils.sha1Hex(pid));
+  }
+
+  @Override
+  public void onCreate(HarvestedResource resource) {
+    String key = DigestUtils.sha1Hex(resource.getUri().toASCIIString());
+    if (resourceNotCreated(key)) {
+      String resourceId = null;
+      try {
+        resourceId = adapter.create(resource);
```

Comment:
❓What should we do when these operations fail? If we just log a debug message and move on, we will be missing events and nobody will even know that resource events failed to be processed. Should we retry? Log an error? @clockard, could we generate some kind of alert here?